numojo/core/datatypes.mojo

```diff
@@ -33,3 +33,146 @@ alias f32 = DType.float32
 """Data type alias for DType.float32"""
 alias f64 = DType.float64
 """Data type alias for DType.float64"""
+
+
+# TODO: Optimize the conditions with dict and move it to compile time
+# Dict can't be created at compile time rn
+struct TypeCoercion:
```

Comment:
I am wondering whether it is better to make a separate file for this struct, e.g., `type_coercion.mojo`.

Reply:
> @forFudan Since Type coercion is part of datatypes, perhaps it's fine to keep it under `datatypes` for now. We could move it later if it grows big.

I agree! let's make it separate in future if needed.